js/interface.js

```diff
@@ -258,7 +258,7 @@ $('#manage-data a').on('click', manageAppData);
 $dataSource.on('change', function onDataSourceListChange() {
   var selectedOption = $(this).find('option:selected');
   var selectedText = selectedOption.text();
-  var selectedValue = selectedOption.val();
+  var selectedValue = parseInt(selectedOption.val(), 10);
```

Comment:
What happens if it's `NaN` ?

Reply:
@hcarneiro same as if this was an empty string, basically won't match any data source https://github.com/Fliplet/fliplet-widget-login-data-source/blob/87459aaaa8fff1db1b00fc9c506b6fda042889fc/js/interface.js#L282-L282